metaflow/plugins/aip/aip_metaflow_step.py

```diff
@@ -35,23 +35,36 @@ def _write_card_artifacts(
     passed_in_split_indexes: str,
     run_id: str,
 ):
+    """
+    Add card artifacts to step outputs. This function depends on Metaflow backend.
+
+    Cards are still available to view through Metaflow UI or Metaflow CLI even if this function fails.
+    Therefore, all exceptions are caught and logged, instead of raised.
+    """
+
     task_id_template: str = f"{task_id}.{passed_in_split_indexes}".strip(".")
     pathspec = f"{flow_name}/{run_id}/{step_name}/{task_id_template}"
 
-    cards: List[Card] = list(get_cards(pathspec))
+    try:
+        cards: List[Card] = list(get_cards(pathspec))
+    except Exception as e:
+        # Workflow should still succeed even if cards fail to render
+        logging.exception(f"Failed to get card artifacts for {pathspec}: {e}")
```

Comment:
Should it fail the workflow?
I understand that get_cards could fail once in a long while querying metaflow metadata service and DB.  One approach to harden this is to add a retry.